live-build/misc/upgrade-scripts/upgrade-container

```diff
@@ -365,25 +378,90 @@ function convert_to_bootfs() {
 	migrate_password_for_user root
 
 	#
-	# LX-72 Until we have a proper solution for migrating the
-	# configuration of a given appliance to the new root filesystem
-	# (i.e. any configuration that isn't supplied by a package), we
-	# resort to explicitly migrating some of this configuration
-	# here. The files listed here are dynamically generated and/or
+	# These files are generic OS files that are required for the
+	# system to properly operate; these must be preserved acorss
+	# upgrades.
+	#
+	# Note, the "/etc/resolv.conf" file is in this list, but the
+	# "migrate_file" function will only migrate this path if it
+	# points to a regular file. Often, this file will be a symlink,
+	# in which case it won't be migrated, and we don't want it to
+	# be. We only want to migrate this file if the admin (or some
+	# software on the system) has converted it to a regular file,
+	# and configured some host specific DNS settings, in which case
+	# we want these settings to persist across the upgrade.
+	#
+	while read -r file; do
+		migrate_file "$file"
+	done <<-EOF
+		/etc/hostname
+		/etc/hosts
+		/etc/machine-id
+		/etc/resolv.conf
+		/etc/ssh/ssh_host_dsa_key
+		/etc/ssh/ssh_host_dsa_key.pub
+		/etc/ssh/ssh_host_ecdsa_key
+		/etc/ssh/ssh_host_ecdsa_key.pub
+		/etc/ssh/ssh_host_ed25519_key
+		/etc/ssh/ssh_host_ed25519_key.pub
+		/etc/ssh/ssh_host_rsa_key
+		/etc/ssh/ssh_host_rsa_key.pub
+		/etc/sudoers.d/90-cloud-init-users
+		/etc/netplan/50-cloud-init.yaml
+	EOF
```

Comment:
Might also want to add `/etc/hostid` and `/etc/zfs/zpool.cache`.

Reply:
what is `/etc/hostid`? I don't see it on my linux system, and a quick grep in the virtualization code leads me to think it's illumos specific..?

Reply:
nvm.. I think I remember, it's created by the zfs commands, right?

Reply:
It is used by ZFS to identify the machine you are on. This file is not there by default on Linux. It's unclear to me if we will need to use it or not, but right now we have to do a work around when importing rpool from Illumos during migration (we must call `zpool import` with `-f`). The hostid property is stored on the pool (you can run `zdb -CC rpool` on Illumos). Note that it's not there on pools created on Linux. The issue is that on Linux, if `/etc/hostid` is not there then the `hostid` command returns some synthetic value based on an IP on your system (for some reason it defaults to the loopback IP, i.e. `007f0101`).